book/standard_library.md

```diff
@@ -14,32 +18,218 @@ The standard library currently includes:
 To see a complete list of the commands available in the standard library, run the following:
 
 ```nu
-use std
-scope commands
-| where name =~ '^std '
-| select name usage extra_usage
-| wrap "std-lib"
+nu -c "
+  use std
+  scope commands
+  | where name =~ '^std '
+  | select name description extra_description
+  | wrap 'Standard Library Commands'
+  | table -e
+"
+```
+
+::: note
+The `use std` command above loads the entire standard library so that you can see all of the commands at once. This is typically not how it will be used (more info below). It is also run in a separate Nu subshell simply so that it is not loaded into scope in the shell you are using.
+:::
+
+## Importing the Standard Library
+
+While working at the commandline, it can be convenient to load the entire standard library using:
+
+```nu
+use std *
+```
+
+However, this form should be avoided in custom commands and scripts since it has the longest load time. While the load time is only on the order of ~20-30 milliseconds on an average system, that can add up in certain scenarios.
+
+::: important Optimal Startup when Using the Standard Library
+See the [notes below](#optimal-startup) on how to ensure that your configuration isn't loading the entire Standard Library.
+:::
+
+### Importing Submodules
+
+Each submodule of the standard library can be loaded separately. Again, for best performance, load only the submodule(s) that you need in your code.
+
+There are several forms that can be used:
```

Comment:
Some of the following points go too much into how modules work, that's not necessary. We can link to https://www.nushell.sh/book/modules.html where it's explained already (although it also needs improvements).

Here you can just say that `std` is a “virtual” directory and the `iter`, `help` are submodules in it, defined as module directories. Linking to https://github.com/nushell/nushell/tree/main/crates/nu-std/std would be helpful to show the module directory structure.

We don't want to teach users to invoke magic incantations. `std` is a normal module like any other module. We shouldn't give an impression that there is anything special about it.

I think you can formulate the points as something like (very abbreviated version):

> The `std` submodules were designed to be used in different ways
> 1. Import the submodule as subcommands
>   Some modules follow a subcommand structure and meant to be imported as `use std/<submodule>`. 
>   a) E.g., `use std/iter` ... defines `iter xxx`subcommands
>   b) Same, but designed to shadow existing commands, e.g., `use std/help`.
>   ... list all modules in this style
> 2. Import the contents of the module directly
>   These modules were designed to make their content available in the current scope with `use std/<submodule> *`.
>   ... list all modules in this style
> 3. Import specific definitions -- IMO this point is not necessary
> 4. Good
>
> When in doubt, checked the source code of the module (linked above).

The terms “prefix” or “namespace” are not very well-defined in Nushell and can be confusing. You can use “scope”, or talk about defining subcommands.

Reply:
I totally agree with you on both points.  (1) We should be able to point to the Modules chapter, and yet (2) the Modules chapter needs a lot of attention.  One of the things we should (i.e., I plan to do) is split out Modules into "Using Modules" and "Writing Modules".  At the moment, a reader would have to wade through *all* of the "writing" information just to find out how to import the different forms.

I'd like to keep this section mostly intact *for now*, and as soon as I get a chance I'll work on the Modules rewrite.  I can use `std` submodules for the examples in the "Using Modules" chapter (or subchapter) since:

1. As you mention `std` is just a standard module/submodule design - Nothing special.
2. It's already on the user's system with no need to come up with "fake examples"
3. I can just *move* (not duplicate) some of this to that chapter where it belongs.

Does that sound like an acceptable path?